### PR TITLE
Update dependencies + test on Node>10 compatible dependencies

### PR DESCRIPTION
**Pull request by aminya**
- This updates all the dependencies 
	- This means that the tests will run on Node>10 compatible dependencies
	- This will unblock other PRs such as #290 that were failing due to old build tools

- Tests threads-js on Node 10-14 via GitHub Actions 

~~- rollup-script and ava-script are used to make it possible to run the tests on old Node versions.~~

### CODE REVIEW

**Comment by aminya** [comment]
Bundle command fails on Node 8 because TypeScript generates import in the esm output and you are trying to bundle the Code from the output of TypeScrpit. Instead, you should use a TypeScript plugin to bundle the code from the beginning.

**Comment by aminya** [comment]
I reverted using Parcel because of https://github.com/parcel-bundler/parcel/issues/5135

**Comment by andywer** [comment]
Side note: You definitely need to keep rollup. Not necessarily for bundling, but just to make sure that our ES module output can be consumed by rollup, since we had issues with rollup in the past 😉

**Comment by aminya** [comment]
Well currently, using the new Rollup people cannot build threads on Node 8. 

Do you want us to use a Pre Node 10 rollup script for that?

**Comment by aminya** [comment]
We can't install two rollups at the same time. Their bin file conflicts! 

Using aliases does not help.
```
    "rollup-old": "npm:rollup@^1",
    "rollup": "^2.26.10",
```

@andywer I think you should decide that you want to be compatible with the old Rollup or the new one (which probably everyone uses).

**Comment by aminya** [comment]
We can use a custom script to call Rollup!

**Comment by aminya** [comment]
The old Node is pretty stupid 💢 It does not even parse `catch {}`.

**Comment by andywer** [comment]
@aminya Thanks so much for your efforts!

> The old Node is pretty stupid 💢 It does not even parse catch {}.

Yeah, the blank `catch {}` syntax hasn't been around for that long. You can of course always write `catch (error) {}` and just not use the `error`, although you might have to tell the linter that this is fine in that case.

I am sorry that I can only contribute so little time at the moment, but I will try to help you as good as I can! That's obviously an important PR.

**Comment by aminya** [comment]
@andywer Could you run CI on this? I fixed the ava issue

**Comment by aminya** [comment]
We need https://github.com/andywer/puppet-run/pull/40

**Comment by andywer** [comment]
@aminya About CI: I am not even sure why it didn't work so far… 🤷‍♂️ 
image

**Comment by andywer** [comment]
@aminya I think the issue with Travis CI stems from force-pushing as the latest build was for the 049d07f commit. Not sure how to fix it, though…

**Comment by aminya** [comment]
I have not seen Travis breaking because of force pushing 🤔

! Well, I tested this offline.

It works perfectly with Node 10 and up.
It still does not work with Node <10.

What I want to suggest is to separate the building of threads from testing it. That means we build threads on Node 10 and above, but we test its runtime for Node <10. Currently, all the build tooling like Parcel, Rollup, Ava, etc have dropped Node <10 support. I can easily write a Github action that builds the code on newer Node and uploads the artifacts for testing on older Node.

**Comment by andywer** [comment]
You know, we can also cut it short. I guess node 8 compatibility isn't a priority anymore. If anyone still needs to stick to such an old node version, they will need to stick to an old threads.js version, too, or at least we won't guarantee newer versions to work with node 8…

Let's drop node 8 from the CI config 😉

**Comment by aminya** [comment]
I dropped Node 8 and Travis 🎉 Here is the running GitHub Action:
https://github.com/aminya/threads.js/actions

**Comment by aminya** [comment]
This is ready to go 🚀

File: package.json [hunk 1]
```diff
@@ -6,18 +6,18 @@
   "main": "dist/index.js",
   "module": "dist-esm/index.js",
   "scripts": {
-    "prebuild": "rimraf dist/ dist-esm/",
-    "build": "run-p build:cjs build:es",
+    "clean": "rimraf dist/ dist-esm/",
+    "build": "npm run clean && npm run build:cjs && npm run build:es",
```
Comment (andywer) [inline package.json:10]: @aminya Any particular reason why you don't want to run it in parallel anymore? It will take much longer this way.

Comment (aminya) [reply package.json:10]: npm-run-all is unmaintained. It randomly fails on Windows. That should not be a problem though. For development, we can add a `dev` command which uses TypeScript's watch feature.

Comment (aminya) [reply package.json:11]: I added dev command in https://github.com/andywer/threads.js/pull/300/commits/c09017e79ee8bea01ee52617284b6f8d97c48a08

**Comment by aminya** [comment]
Here is the CI run for the record: https://github.com/aminya/threads.js/actions/runs/333457135

Not sure why it is not shown here.

File: package.json (hunk 1 above)
Comment (andywer) [reply package.json:11]: Fair enough, but how about [`concurrently`](https://www.npmjs.com/package/concurrently) then?

**Comment by andywer** [comment]
@aminya Btw, I really appreciate the time and effort you put into this! Especially since I don't have much time right now. Great job 👍

**Comment by aminya** [comment]
You're welcome 👍🏻 😉

File: package.json (hunk 1 above)
Comment (aminya) [reply package.json:11]: We can try that.

**Comment by andywer** [comment]
@aminya There is one thing about the GitHub actions workflow that we still need to fix: When the test fails with the retry action, we do not get to see the original error anymore: [see here, for instance](https://github.com/aminya/threads.js/runs/1320374232?check_suite_focus=true).

But I guess we should not auto-retry flaky tests in the first place. Let's check why they are flaky and fix them. It's also a bit strange, since they weren't flaky before – I could run them reliably on Travis CI and locally 🤔

**Comment by aminya** [comment]
> @aminya There is one thing about the GitHub actions workflow that we still need to fix: When the test fails with the retry action, we do not get to see the original error anymore: [see here, for instance](https://github.com/aminya/threads.js/runs/1320374232?check_suite_focus=true).
> 
> But I guess we should not auto-retry flaky tests in the first place. Let's check why they are flaky and fix them. It's also a bit strange, since they weren't flaky before – I could run them reliably on Travis CI and locally 🤔

The tests of the package itself are fine. The test that bundles the code using Rollup sometimes timeouts! I tried to increase its timeout number in [this commit](https://github.com/andywer/threads.js/pull/300/commits/31594e6687269df718adc0d3c2182737270dc302), but that does not seem to make a difference 🤷‍♂️ Maybe I am doing it wrong.

**Comment by andywer** [comment]
> The test that bundles the code using Rollup sometimes timeouts!

Ohh, I see… Maybe we have a race condition in the rollup testing code. I can look into it later – some additional logging might give us a hint.

**Comment by aminya** [comment]
Bump

**Comment by andywer** [comment]
Thanks for the reminder! No time to look into it today, though, I fear… Maybe I get around looking into it until weekend.

**Comment by aminya** [comment]
Thanks. Hopefully, we can merge this on the weekend. This was ready and working. But some stuff has merged into master.

**Comment by andywer** [comment]
Merged 😉

**Comment by andywer** [comment]
@aminya Thanks again! Btw, do you need an update published urgently or were you just waiting for the dev dependencies to get fixed?

**Comment by aminya** [comment]
@andywer Thanks! This was blocking #290. Now I can merge that PR.

**Pull request merged**